src/channel/ChannelContext.cpp

```diff
@@ -255,7 +255,16 @@ void ChannelContext::HandleNodeIdResolve(CHIP_ERROR error, uint64_t nodeId, cons
 CHIP_ERROR ChannelContext::SendSessionEstablishmentMessage(const PacketHeader & header, const Transport::PeerAddress & peerAddress,
                                                            System::PacketBufferHandle msgIn)
 {
-    return mExchangeManager->GetSessionMgr()->GetTransportManager()->SendMessage(header, peerAddress, std::move(msgIn));
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    err = header.EncodeBeforeData(msgIn);
```

Comment:
nit: I think we try to use `ReturnErrorOnFailure` instead of `SuccessOrExit` if there's nothing to be done in the `exit` block.

Reply:
In general, I agree. But it is still case by case. Those macros are inherited from Openweave, and in which ReturnErrorOnFailure is used to wrap simple expressions for fast return.

For a function with a lot of complicated expressions, if we use ReturnErrorOnFailure for each of them, this will make the code more difficult to read. Such as:

ReturnErrorOnFailure(mExchangeManager->GetSessionMgr()->GetTransportManager()->SendMessage(header, peerAddress, std::move(msgIn)));

Reply:
Why not just call `ReturnErrorOnFailure(err);`?

Reply:
Acked